### PR TITLE
Unbroken

### PR DESCRIPTION
**Pull request by crhym3**
This fixes the `broken` branch and also cleans up the acme.go code in general.
All funcs are now methods of a `Client`, which looks nice and easy to use, imho.

R: @sgomes, @surma

### CODE REVIEW

File: acme.go [hunk 1]
```diff
@@ -66,149 +83,119 @@ type ChallengeSet struct {
...
 	}
-	return nil
+	return cert, res.Header.Get("Location"), nil
 }
 
-// Register create a new registration by following the "new-reg" flow,
-// using the provided config. The config must have a non-nil Key field
-// and a non-nil Endpoint.RegURL.
+// Register create a new registration by following the "new-reg" flow.
+// It populates a argument with the response received from the server.
```
Comment (sgomes) [inline acme.go]: Nit: "It populates the `a` argument"

File: acme.go [hunk 2]
```diff
@@ -66,149 +83,119 @@ type ChallengeSet struct {
...
+	return a, c.doReg(url, a, "reg")
 }
 
-// UpdateReg retrieves an existing registration, using the provided config.
-// The config must have non-nil Key and RegURI fields.
-//
-// Config fields will be updated with the server response.
-// TODO: describe what gets updated.
+// UpdateReg updates existing registration.
+// It populates a argument with the response received from the server.
```
Comment (sgomes) [inline acme.go]: Same as above.

**Comment by sgomes** [comment]
LGTM, minor nit. Feel free to merge after that.

File: acme.go (hunk 1 above)
Comment (crhym3) [inline acme.go]: done, w/o quotes though. it's not common to use quotes for args.

File: acme.go (hunk 2 above)
Comment (crhym3) [inline acme.go]: done

**Pull request merged**